docs/md/expressions.md

```diff
@@ -0,0 +1,429 @@
+
+### Ternary Operators
+
+This can be used to create a value based on a condition. It may be used anywhere
+a value is expected, such as a variable assignment or a return value.
+
+The format is `CONDITION ? VALUE_WHEN_TRUE : VALUE_WHEN_FALSE;`.
+
+```html
+// My Column Name
+
+var priceAdjustmentDate := date(2016, 6, 18);
+var initialPrice := "Sales" - "Discount";
+
+"Order Date" > priceAdjustmentDate ? initialPrice - 5 : initialPrice + 5;
+```
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Discount", "Order Date", "My Column Name"]'
+  expressions='["// My Column Name\n\nvar priceAdjustmentDate := date(2016, 6, 18);\nvar initialPrice := \"Sales\" - \"Discount\";\n\n\"Order Date\" > priceAdjustmentDate ? initialPrice - 5 : initialPrice + 5;"]'
+></perspective-viewer>
+</div>
+
+### `switch case` Block
+
+Run different commands based on any number of exclusive conditions. The optional
+`default` condition is taken when no other conditions are met.
+
+```html
+// My Column Name
+
+var priceAdjustmentDate := date(2015, 10, 11);
+var price := "Sales" - "Discount";
+
+switch {
+  case "Order Date" > priceAdjustmentDate:
+    price -= 5;
+  case "Order Date" < priceAdjustmentDate:
+    price += 5;
+  default:
+    price := max(price - 100, 0); // Special discount on day of adjustment.
+}
+
+price;
+```
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Discount", "Order Date", "My Column Name"]'
+  expressions='["// My Column Name\n\nvar priceAdjustmentDate := date(2015, 10, 11);\nvar price := \"Sales\" - \"Discount\";\n\nswitch {\n  case \"Order Date\" > priceAdjustmentDate:\n    price -= 5;\n  case \"Order Date\" < priceAdjustmentDate:\n    price += 5;\n  default:\n    price := max(price - 100, 0); // Special discount on day of adjustment.\n}\n\nprice;"]'
+></perspective-viewer>
+</div>
+
+## Loops
+
+Repeat the same actions for aggregate effect.
+
+### `for` Loop
+
+The basic form of this loop is such:
+
+```html
+for (var x := 0; x < NUMBER_OF_REPETITIONS; x += 1) {
+  // COMMANDS
+}
+```
+
+#### `for` Example
+
+```html
+// My Column Name
+
+var count := 0;
+for (var x := 0; x < 5; x += 1) {
+  count += 1;
+}
+
+count;
+```
+
+<div>
+<perspective-viewer
+  columns='["Sales", "Discount", "Order Date", "My Column Name"]'
+  expressions='["// My Column Name\n\nvar count := 0;\nfor (var x := 0; x < 5; x += 1) {\n  count += 1;\n}\n\ncount;"]'
+></perspective-viewer>
+</div>
+
+<!-- TODO: Could not make this work. -->
+<!-- ### `while` Loop -->
+
+<!-- TODO: Could not make this work. -->
+<!-- ### `repeat until` Loop -->
+
+<!-- TODO: Could not make this work. -->
+<!-- ### `break` -->
+
+<!-- TODO: Could not make this work. -->
+<!-- ### `continue` -->
+
+## Advanced Example
+
+This **[Mandelbrot
+demo](https://bl.ocks.org/texodus/5485f6b630b08d38218822e507f09f21)** calculates
+the popular fractal in a Column Expression and displays it with a heatmap. Edit
+the `color` column to view and manipulate the expression, whose contents are
+below:
```

Comment:
I really like the idea of having more complext PerspExprTK examples in the docs, but this one is both a little complicated and lacking all of the detail needed to actually implement it (e.g. source `Table` structure, X/Y axis calculations).